comp.go

```diff
@@ -74,15 +79,12 @@ func equals(a, b any) bool {
 		return a == b
 	}
 
-	if isNumber(a) {
-		return isPrimitive(b) && toNumber(a) == toNumber(b)
+	if isPrimitive(a) && isPrimitive(b) && (isNumber(a) || isNumber(b)) {
+		return toNumberForLess(a) == toNumberForLess(b)
 	}
 
-	if isBool(a) {
-		if !isBool(b) {
-			return false
-		}
-		return isTrue(a) == isTrue(b)
+	if isBool(a) || isBool(b) {
+		return isTruthy(a) == isTruthy(b)
```

Comment:
Why not use `isTrue` here?

Reply:
see https://github.com/jwadhams/json-logic/pull/48 a lot of test cases that fail that this PR is fixing